src/navigation/app/app-navigation.tsx

```diff
@@ -24,47 +30,60 @@ const screenOptions: NativeStackNavigationOptions = {
 const AppNavigation = () => {
   const styles = useStyles();
   const black = useColor('BLACK');
+  const { user } = useAppSelector(state => state.auth);
+
+  const isUserAdmin = user?.role === UserRoleType.Admin;
```

Comment:
Add useMemo.

Reply:
I think useMemo is not needed here, because it will do the same check, but also take up memory